tests/cli.rs

```diff
@@ -660,3 +660,122 @@ fn cert_with_key() {
         .stdout(predicates::str::contains("client-authenticated"))
         .stderr(predicates::str::is_empty());
 }
+
+#[test]
+fn forced_json() {
+    let server = MockServer::start();
+    let mock = server.mock(|when, _then| {
+        when.method(GET)
+            .header("content-type", "application/json")
+            .header("accept", "application/json, */*;q=0.5");
+    });
+    get_command()
+        .arg("--json")
+        .arg(server.base_url())
+        .assert()
+        .success();
+    mock.assert();
+}
+
+#[test]
+fn forced_form() {
+    let server = MockServer::start();
+    let mock = server.mock(|when, _then| {
+        when.method(GET)
+            .header("content-type", "application/x-www-form-urlencoded");
+    });
+    get_command()
+        .arg("--form")
+        .arg(server.base_url())
+        .assert()
+        .success();
+    mock.assert();
+}
+
+#[test]
+fn forced_multipart() {
+    let server = MockServer::start();
+    let mock = server.mock(|when, _then| {
+        when.method(POST).header_exists("content-type").body("");
+    });
+    get_command()
+        .arg("--multipart")
+        .arg(server.base_url())
+        .assert()
+        .success();
+    mock.assert();
+}
+
+#[test]
+fn formatted_json_output() {
+    let server = MockServer::start();
+    let mock = server.mock(|_when, then| {
+        then.header("content-type", "application/json")
+            .body(r#"{"":0}"#);
+    });
+    get_command()
+        .arg("--print=b")
+        .arg(server.base_url())
+        .assert()
+        .stdout(indoc! {r#"
+        {
+            "": 0
+        }
+
+        "#});
+    mock.assert();
+}
+
+#[test]
+fn inferred_json_output() {
+    let server = MockServer::start();
+    let mock = server.mock(|_when, then| {
+        then.header("content-type", "text/plain").body(r#"{"":0}"#);
+    });
+    get_command()
+        .arg("--print=b")
+        .arg(server.base_url())
+        .assert()
+        .stdout(indoc! {r#"
+        {
+            "": 0
+        }
+
+        "#});
+    mock.assert();
+}
+
+#[test]
+fn inferred_nonjson_output() {
+    let server = MockServer::start();
+    let mock = server.mock(|_when, then| {
+        // Trailing comma makes it invalid JSON, though formatting would still work
+        then.header("content-type", "text/plain").body(r#"{"":0,}"#);
+    });
+    get_command()
+        .arg("--print=b")
+        .arg(server.base_url())
+        .assert()
+        .stdout(indoc! {r#"
+        {"":0,}
+        "#});
+    mock.assert();
+}
+
+#[test]
+fn noninferred_json_output() {
+    let server = MockServer::start();
+    let mock = server.mock(|_when, then| {
+        // Trailing comma makes it invalid JSON, though formatting would still work
+        then.header("content-type", "application/octet-stream")
+            .body(r#"{"":0}"#);
```

Comment:
I don't see a trailing comma here.

Reply:
That's what I get for copy-pasting :stuck_out_tongue: